com.unity.shadergraph/Editor/Data/Graphs/Vector2MaterialSlot.cs

```diff
@@ -70,7 +75,14 @@ public Vector2 value
 
         public override VisualElement InstantiateControl()
         {
-            return new MultiFloatSlotControlView(owner, labels, () => value, (newValue) => value = newValue);
+            if (m_Integer)
+            {
+                return new MultiIntegerSlotControlView(owner, labels, () => value, (newValue) => value = newValue);
```

Comment:
Note that this only affects the default slot UI -- enforcing that users can only type integers into those default boxes that hang off the node.  The values are still stored as floats, passed as floats, and the user can hook up a Vector2 Node that passes non-integer values.  Is that intended?

Reply:
Yep, you can certainly force a float value through via another node, and actually the shader always receives a float. It's really for intuitive setting of the offset values to Gather via the slot, which takes in int2 inside the actual shader, when you use the intrinsic function. The fallback path could probably be forced to use a non-integer offset if I don't put a round in there.

Reply:
I made sure that the fallback path uses an integer value for the texture sampling, to always match the use of the actual Gather function (which always takes an integer value).